### PR TITLE
remove deprecated resource and datasource

### PR DESCRIPTION
**Pull request by duytiennguyen-okta**
remove deprecated resource and data source as noted in https://github.com/okta/terraform-provider-okta/issues/1338
I also removed some of attribute that was removed in the resource but not the data source ([data_source_okta_user.go](https://github.com/okta/terraform-provider-okta/pull/1532/files#diff-1370d9b45a7d38f8d662649cf9d364d159c0bc0d698c98ecb862f229b4828da2))
These 2 test still fail `TestAccOktaPolicyRuleIdpDiscovery_crud` and `TestAccOktaUser_customProfileAttributes`

### CODE REVIEW

File: okta/resource_okta_policy_rule_idp_discovery_test.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
+// TODU
```
Comment (monde) [inline okta/resource_okta_policy_rule_idp_discovery_test.go:10]: Is there still a TODO here?

File: okta/resource_okta_user_test.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
+// TODU
```
Comment (monde) [inline okta/resource_okta_user_test.go:18]: Done?

**Review by monde — APPROVED** [review]
What a liberating PR! `+85 −2,382`
:shipit:

**Comment by virgofx** [comment]
Super excited about removing all the cruft!

**Comment by virgofx** [comment]
Hey @duytiennguyen-okta Any info on when we can expect a 4.x release? Would be happy to test.

**Comment by duytiennguyen-okta** [comment]
@virgofx It should be release by the end of the week

**Pull request merged**